### PR TITLE
fix: ensure we are talking to coder on first user check

### LINKED ISSUE #11006: feat(cli): (login) Improve invalid URL experience
When using `coder login <URL>` on a new deployment, coder will always prompt you to create the first user, even though the URL might not even be a Coder instance.
This occured to me when improperly setting up my reverse-proxy and thus, making the Coder instance unreachable, but the CLI still "thought" it was a Coder instance.

---

```
PS C:\Users\Phorcys> coder login https://google.com
build info: GET https://google.com/api/v2/buildinfo: unexpected status code 404: unexpected non-JSON response "text/html; charset=UTF-8"
        Error: <!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 404 (Not Found)!!1</title>
  <style>
    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}
  </style>
  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>
  <p><b>404.</b> <ins>That’s an error.</ins>
  <p>The requested URL <code>/api/v2/buildinfo</code> was not found on this server.  <ins>That’s all we know.</ins>

> Your Coder deployment hasn't been set up!
> Would you like to create the first user? (yes/no)
```

---

`Coder v2.4.0+d19a762`

---

cc: @matifali

### PR DESCRIPTION
**Pull request by f0ssel**
Closes https://github.com/coder/coder/issues/11006

### CODE REVIEW

**Comment by Emyrk** [comment]
Can we check for the `x-coder-build-version` header instead of a body string?

**Comment by f0ssel** [comment]
Good call @Emyrk I have updated it

**Review by Emyrk — COMMENTED** [review]
We should add a small unit test probably, and mentioned something about the error that I'll leave up to you.

File: codersdk/users.go [hunk 1]
```diff
@@ -190,7 +190,15 @@ func (c *Client) HasFirstUser(ctx context.Context) (bool, error) {
 		return false, err
 	}
 	defer res.Body.Close()
+
 	if res.StatusCode == http.StatusNotFound {
+		// ensure we are talking to coder and not
+		// some other service that returns 404
+		v := res.Header.Get("X-Coder-Build-Version")
+		if v == "" {
+			return false, xerrors.Errorf("missing build version header, not a coder instance")
```
[the remark this answers is not in the document]
Comment (f0ssel) [reply codersdk/users.go:199]:
```
Failed to check server "https://google.com" for first user, is the URL correct and is coder accessible from your browser? Error - has initial user: missing build version header, not a coder instance```

[the remark this answers is not in the document]
Comment (Emyrk) [reply codersdk/users.go:199]: Perfect!

**Review by Emyrk — APPROVED** [review]

**Pull request merged**